scss/_icon.scss

```diff
@@ -0,0 +1,100 @@
+/// Icon Buttons
+///
+/// Outputs styles for every icon in o-buttons-icons at every theme in o-buttons-themes
+
+@mixin oButtonsIcon($buttonClass: $o-buttons-class) {
+  // Browserhack to only apply these styles for IE7 and up. IE6 will get the
+  // text fallback. Hack documented here: http://browserhacks.com/#hack-da690292d4fddd94dc7bdd50e38b5713
+  html > body .#{$buttonClass}-icon {
+    display: inline-block;
+    background-repeat: no-repeat;
+    background-position: 3px;
+    padding-left: 22px;
+
+    &.o-buttons-icon--icon-only {
+      padding-left: 0px;
+      background-position: 50%;
+    }
+
+    .#{$buttonClass}-icon__label {
+      @include _oButtonsIconButtonLabel();
+    }
+
+    @each $theme, $properties in $o-buttons-themes {
+      @each $icon in $o-buttons-icons {
+        @include oButtonsGetButtonForIconAndTheme($icon, $theme);
+      }
+    }
+  }
+}
+
+/// Get Button For Icon and Theme
+///
+/// Outputs background-image property for a given icon/theme at all states set
+/// in that theme's state list
+/// example:
+/// .my-button--left-arrow {
+///   @include oButtonsGetButtonForIconAndTheme(left-arrow, standout);
+/// }
+///
+/// @param {String} $icon-name, any icon name found in o-ft-icons
+/// @param {String} $theme, any theme name as defined in $o-buttons-themes (standard, standout, etc). Defaults to standard.
+/// @param {String} $button-class, defaults to o-buttons ($o-buttons-class' default value)
+///
+
+@mixin oButtonsGetButtonForIconAndTheme($icon-name, $theme: standard, $buttonClass: $o-buttons-class) {
+
+  &.#{$buttonClass}-icon--#{$icon-name} {
+
+    @if $theme == 'standard' {
+      $theme-selector: '&';
+    } @else {
+      $theme-selector: '&.#{$buttonClass}--#{$theme}';
+    }
+
+    #{$theme-selector} {
+      $state-list: map-get($o-buttons-themes, $theme); // normal, active, hover, focus, pressedselected, disabled
+
+      @include _oButtonsGetIconForThemeAndState($icon-name, $state-list, normal);
+
+      // http://www.w3.org/TR/wai-aria/states_and_properties#aria-selected
+      // http://www.w3.org/TR/wai-aria/states_and_properties#aria-pressed
+	    &[aria-selected=true], // e.g. A selected tab or page number in pagination
+	    &[aria-pressed=true] { // e.g. A "follow" button that is pressed
+        @include _oButtonsGetIconForThemeAndState($icon-name, $state-list, pressedselected);
+      }
+
+      &[disabled] {
+        @include _oButtonsGetIconForThemeAndState($icon-name, $state-list, disabled);
+      }
+
+      &:not([disabled]):active {
+        @include _oButtonsGetIconForThemeAndState($icon-name, $state-list, active);
+      }
+
+      // Hack to get the active state colour svg to download to prevent FOIC
+      &:after {
+        @include _oButtonsGetIconForThemeAndState($icon-name, $state-list, active);
+        content: '';
```

Comment:
Oh, of course!

-- 

------------------------------

*This email was sent by a company owned by Financial Times Group Limited 
("FT Group <http://aboutus.ft.com/corporate-information/#axzz3rajCSIAt>"), 
registered office at Number One Southwark Bridge, London SE1 9HL.  
Registered in England and Wales with company number 879531. This e-mail may 
contain confidential information. If you are not the intended recipient, 
please notify the sender immediately, delete all copies and do not 
distribute it further.  It could also contain personal views which are not 
necessarily those of the FT Group.  We may monitor outgoing or 
incoming emails as permitted by law.*